charts/gateway-helm/templates/envoy-proxy-topology-mutator-webhook.yaml

```diff
@@ -0,0 +1,37 @@
+{{- if .Values.topologyWebhook.enabled }}
+{{- $secret := default (dict "data" (dict)) (lookup "v1" "Secret" .Release.Namespace "envoy-gateway") }}
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: envoy-gateway-topology-webhook
+  {{- with .Values.topologyWebhook.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  labels:
+    app.kubernetes.io/component: topology-webhook
+  {{- include "eg.labels" . | nindent 4 }}
+webhooks:
+  - name: topology.webhook.envoyproxy.io
+    admissionReviewVersions: ["v1"]
+    sideEffects: NoneOnDryRun
+    timeoutSeconds: 5
+    clientConfig:
+      caBundle: {{ default "" (index $secret.data "ca.crt" ) | b64dec }}
+      service:
+        name: {{ include "eg.webhookService" . }}
+        namespace: '{{ .Release.Namespace }}'
+        path: "/mutate-pod-topology"
+        port: 443
+    rules:
```

Comment:
is there a way to add a label to the proxy pod and filter/match here ?

Reply:
No because pod labels/annotations aren't copied to the relevant Binding resource so the filtering happens inside the hook.

Reply:
thanks, resolving this one